microsoft/testsuites/network/sriov.py

```diff
@@ -542,173 +529,360 @@ def sriov_ethtool_offload_setting(self, environment: Environment) -> None:
+                    " after iperf3 run!"
+                ).is_greater_than(init_interrupts_value)
+
     def after_case(self, log: Logger, **kwargs: Any) -> None:
         environment: Environment = kwargs.pop("environment")
         for node in environment.nodes.list():
             kill = node.tools[Kill]
             kill.by_name("iperf3")
 
-    def _initialize_nic_info(
-        self, environment: Environment
-    ) -> Dict[str, Dict[str, NicInfo]]:
-        vm_nics: Dict[str, Dict[str, NicInfo]] = {}
-        for node in environment.nodes.list():
-            node_nic_info = Nics(node)
-            node_nic_info.initialize()
-            for _, node_nic in node_nic_info.nics.items():
-                assert_that(node_nic.lower).described_as(
-                    f"This interface {node_nic.upper} does not have a paired VF."
-                ).is_not_empty()
-            vm_nics[node.name] = node_nic_info.nics
-        return vm_nics
-
-    def _remove_module(self, node: Node) -> str:
-        lspci = node.tools[Lspci]
-        modprobe = node.tools[Modprobe]
-        devices_slots = lspci.get_devices_slots(
-            constants.DEVICE_TYPE_SRIOV, force_run=True
-        )
-        # there will not be multiple Mellanox types in one VM
-        # get the used module using any one of sriov device
-        module_in_used = lspci.get_used_module(devices_slots[0])
-        assert_that(self.reload_modules_dict).described_as(
-            f"used modules {module_in_used} should be contained"
-            f" in dict {self.reload_modules_dict}"
-        ).contains(module_in_used)
-        modprobe.remove(self.reload_modules_dict[module_in_used])
-        return module_in_used
-
-    def _load_module(self, node: Node, module_name: str) -> None:
-        modprobe = node.tools[Modprobe]
-        modprobe.load(module_name)
-
-    def _get_packets(self, node: Node, nic_name: str, name: str = "tx_packets") -> int:
-        cat = node.tools[Cat]
-        return int(
-            cat.read_from_file(
-                f"/sys/class/net/{nic_name}/statistics/{name}", force_run=True
-            )
-        )
 
-    def _sriov_basic_test(
-        self, environment: Environment, vm_nics: Dict[str, Dict[str, NicInfo]]
-    ) -> None:
-        for node in environment.nodes.list():
-            # 1. Check module of sriov network device is loaded.
-            modules_exist = False
-            lsmod = node.tools[Lsmod]
-            for module in ["mlx4_core", "mlx4_en", "mlx5_core", "ixgbevf"]:
-                if lsmod.module_exists(module):
-                    modules_exist = True
-            assert_that(modules_exist).described_as(
-                "The module of sriov network device isn't loaded."
-            ).is_true()
-
-            # 2. Check VF counts listed from lspci is expected.
-            lspci = node.tools[Lspci]
-            devices_slots = lspci.get_devices_slots(
-                constants.DEVICE_TYPE_SRIOV, force_run=True
+@TestSuiteMetadata(
```

Comment:
The stress case can be put in a separated file. It's clearer.

Reply:
updated